src/cluster_resources.rs

```diff
@@ -41,15 +50,132 @@ pub trait ClusterResource:
     + GetApi<Namespace = str>
     + Serialize
 {
+    fn maybe_mutate(self, _strategy: &ClusterResourceApplyStrategy) -> Self {
+        self
+    }
+}
+
+#[derive(Debug, Display, Eq, PartialEq)]
+pub enum ClusterResourceApplyStrategy {
+    /// Default strategy. Resources a applied via the [`Client::apply_patch`] client method.
+    Default,
+    /// Strategy to pause reconciliation. This means any cluster changes (e.g. in the custom resource
+    /// spec) are ignored. Resources are not applied at all but merely fetched via the
+    /// [`Client::get`] method in order to still be able to e.g. update the cluster status.
+    ReconciliationPaused,
+    /// Strategy to stop the cluster. This means no Pods should be scheduled for either [`StatefulSet`],
+    /// [`DaemonSet`] or [`Deployment`]. This is done by setting the [`StatefulSet`] or [`Deployment`]
+    /// replicas to 0. For [`DaemonSet`] this is done via an unreachable [`NodeSelector`].
+    ///
+    /// Resources are applied via the [`Client::apply_patch`] client method.
+    ClusterStopped,
+    /// Dry-run strategy that doesn't actually create any workload resources. This is useful for
+    /// Superset and Airflow clusters that need to wait for their databases to be set up first.
+    NoApply,
+}
+
+impl From<&ClusterOperation> for ClusterResourceApplyStrategy {
+    fn from(commons_spec: &ClusterOperation) -> Self {
```

Comment:
This is just a helper for less code. You do not have to use this when defining the strategy. 
The problem would be if you want another strategy you would have to add one in the operator-rs. 
But currently i dont see a way around this or actually having a priority since these (and additional) flags are intertwined.

Reply:
My (completely not thought through) idea was along the lines of making [this](https://github.com/stackabletech/operator-rs/pull/573/files/ccc22816beebcb8d6b216ae1380a965675dab7d9#diff-c7e545facebe791742ac9b2a316e211d02158fda0a643655d60a62c54b316327R320) a `Vec<ClusterResourceApplyStrategy>` and then `maybe_mutate`ing all of them in turn when the time comes. That would allow us a degree of freedom if we ever add unrelated strategies (the label one actually might be a useful one).

We'd still need to add the strategy in operator-rs and that is totally fine, but we'd have the option of using more than one at the same time.

But again, not thought out and I only wanted to put it out there in case it might avoid a breaking api change down the road.

Reply:
I think the requirement  valid but the proposed implementation is not possible because a strategy bundles to things:
1. How to (possibly) mutate a resource before applying it.
2. How to send it (or not send it) to the K8S api.
The second point is where things break down badly if multiple strategies are allowed.

To implement this feature, a larger redesign is needed to separate the two concerns.